### PR TITLE
feat(workflow): Handle data to performance section on release details

### PR DESCRIPTION
**Pull request by mikellykels**
Handle data on the Performance Section on the Release Details page for Frontend, Backend, Mobile, and Unknown.

https://www.figma.com/file/jsZJ9v8ditpm4uBISafH9P/Releases-V4?node-id=107%3A4721

[FIXES WOR-1391](https://getsentry.atlassian.net/browse/WOR-1391)
[FIXES WOR-967](https://getsentry.atlassian.net/browse/WOR-967)

### Frontend
Screen Shot 2021-11-01 at 10 21 30 PM

### Backend
Screen Shot 2021-10-19 at 11 11 51 PM

### Mobile
Screen Shot 2021-10-25 at 9 49 35 PM

### Unknown
Screen Shot 2021-10-19 at 10 56 01 PM

### Dark Mode
Screen Shot 2021-11-01 at 11 56 03 PM

### "No results found" (can be either this release or all releases)
Screen Shot 2021-11-01 at 11 34 38 PM
Screen Shot 2021-10-25 at 9 49 03 PM

### CODE REVIEW

File: static/app/components/discover/performanceCardTable.tsx [hunk 1]
```diff
@@ -1,100 +1,225 @@
...
+        <SubTitle key={idx}>
+          <Link
+            to={{
+              pathname: discoverPath,
+              query: {
+                query: `${releaseEventView.query}`,
+                project: `${releaseEventView.project}`,
+                field: `${span.column}`,
+                statsPeriod: `${releaseEventView.statsPeriod}`,
+              },
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:150]: You should be able to use `newView.getResultsViewUrlTarget(org.slug)` to do this. You will have to modify the event view first with `const newView = releaseEventView.withColumns()` to get a new view with just the column you're interested in showing.

File: static/app/components/discover/performanceCardTable.tsx [hunk 2]
```diff
@@ -1,100 +1,225 @@
...
+          data: thisReleaseTableData?.data?.[0]?.[span.field],
+          meta: thisReleaseTableData?.meta?.[span.field],
+        },
+      };
+    });
+
+    const emptyColumn = (
+      <div>
+        <SingleEmptySubText>
+          <StyledNotAvailable tooltip="No results found" />
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:199]:
```suggestion
          <StyledNotAvailable tooltip={t('No results found')} />
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 3]
```diff
@@ -1,100 +1,225 @@
...
+            <MultipleEmptySubText key={vital[index]}>
+              {<StyledNotAvailable tooltip="No results found" />}
+            </MultipleEmptySubText>
+          ))}
+        </StyledPanelItem>
+        <StyledPanelItem>
+          <TitleSpace />
+          {spans.map((span, index) => (
+            <MultipleEmptySubText key={span[index]}>
+              {<StyledNotAvailable tooltip="No results found" />}
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:213]:
```suggestion
              {<StyledNotAvailable tooltip={t('No results found')} />}
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 4]
```diff
@@ -105,62 +230,174 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
-        </Field>
+        <SubTitle key={idx}>
+          <Link
+            to={{
+              pathname: discoverPath,
+              query: {
+                query: `${releaseEventView.query}`,
+                project: `${releaseEventView.project}`,
+                field: `${span.column}`,
+                statsPeriod: `${releaseEventView.statsPeriod}`,
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:334]: You could use the EventView methods here too.

File: static/app/components/discover/performanceCardTable.tsx [hunk 5]
```diff
@@ -105,62 +230,174 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
-
+    }
+
+    const emptyColumn = (
+      <div>
+        <SingleEmptySubText>
+          <StyledNotAvailable tooltip="No results found" />
+        </SingleEmptySubText>
+        <SingleEmptySubText>
+          <StyledNotAvailable tooltip="No results found" />
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:386]:
```suggestion
          <StyledNotAvailable tooltip={t('No results found')}/>
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 6]
```diff
@@ -105,62 +230,174 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
+        </SingleEmptySubText>
+        <StyledPanelItem>
+          <TitleSpace />
+          {spans.map((span, index) => (
+            <MultipleEmptySubText key={span[index]}>
+              {<StyledNotAvailable tooltip="No results found" />}
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:392]:
```suggestion
              {<StyledNotAvailable tooltip={t('No results found')} />}
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 7]
```diff
@@ -170,170 +407,349 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
-        <div>
-          {this.userMiseryField()}
-          {field}
-        </div>
-      );
-    };
+    const emptyColumn = (
+      <div>
+        <SingleEmptySubText>
+          <StyledNotAvailable tooltip="No results found" />
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:517]:
```suggestion
          <StyledNotAvailable tooltip={t('No results found')} />
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 8]
```diff
@@ -170,170 +407,349 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
       </Fragment>
     );
   }
 
-  renderUnknownPerformance() {
+  function renderUnknownPerformance() {
+    const emptyColumn = (
+      <div>
+        <SingleEmptySubText>
+          <StyledNotAvailable tooltip="No results found" />
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:593]:
```suggestion
          <StyledNotAvailable tooltip={t('No results found')} />
```

File: static/app/components/discover/performanceCardTable.tsx [hunk 9]
```diff
@@ -170,170 +407,349 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
-      ? this.renderBackendPerformance()
-      : MOBILE_PLATFORMS.includes(project.platform as string)
-      ? this.renderMobilePerformance()
-      : this.renderUnknownPerformance();
-    const isUnknownPlatform = ![
-      ...FRONTEND_PLATFORMS,
-      ...BACKEND_PLATFORMS,
-      ...MOBILE_PLATFORMS,
-    ].includes(project.platform);
+  const loader = <LoadingIndicator style={{margin: '70px auto'}} />;
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:636]: You could use emotion to apply this style.

File: static/app/components/discover/performanceCardTable.tsx [hunk 10]
```diff
@@ -170,170 +407,349 @@ class PerformanceCardTable extends React.PureComponent<Props> {
...
+    <Fragment>
+      <HeadCellContainer>
+        {platformPerformanceRender[performanceType].title}
+      </HeadCellContainer>
+      {isUnknownPlatform && (
+        <StyledAlert type="warning" icon={<IconWarning size="md" />} system>
+          For more performance metrics, specify which platform this project is using in{' '}
+          <Link to={`/settings/${organization.slug}/projects/${project.slug}/`}>
+            project settings.
+          </Link>
```
Comment (markstory) [inline static/app/components/discover/performanceCardTable.tsx:669]: Should this have a translation?

File: static/app/views/performance/utils.tsx [hunk 11]
```diff
@@ -34,13 +40,15 @@ const BACKEND_PLATFORMS: string[] = [...backend];
...
-  projects: Project[],
+  projects: (Project | ReleaseProject)[],
   projectIds: readonly number[]
 ) {
   if (projectIds.length === 0 || projectIds[0] === ALL_ACCESS_PROJECTS) {
     return PROJECT_PERFORMANCE_TYPE.ANY;
   }
-  const selectedProjects = projects.filter(p => projectIds.includes(parseInt(p.id, 10)));
+  const selectedProjects = projects.filter(p =>
+    projectIds.includes(parseInt(`${p.id}`, 10))
```
Comment (markstory) [inline static/app/views/performance/utils.tsx:50]: Isn't p.id already a string?

**Review by markstory — APPROVED** [review]
Left a few comments, nothing that I feel should be blocking though as I assume this view will continue to evolve as you get more feedback.

**Review by scttcper — APPROVED** [review]

File: static/app/views/performance/utils.tsx (hunk 11 above)
Comment (mikellykels) [reply static/app/views/performance/utils.tsx:50]: ah it is for the `Project` type, but `ReleaseProject.id` is a `number`, and typescript was complaining here about type`number` is not assignable to type `string`

**Pull request merged**